jax/_src/numpy/lax_numpy.py

```diff
@@ -1106,7 +1104,12 @@ def where(condition: ArrayLike, x: ArrayLike | None = None,
   else:
     util.check_arraylike("where", condition, x, y)
     if size is not None or fill_value is not None:
-      raise ValueError("size and fill_value arguments cannot be used in three-term where function.")
+      raise ValueError("size and fill_value arguments cannot be used in "
+                       "three-term where function.")
+    if x is None or y is None:
```

Comment:
This error is checked inside `util._where`, however its type annotation forbids calling it in this condition, so the type checker demands that we check this outside.